packages/legacy-client/src/ontology-runtime/ontologyProvider/OntologyProvider.ts

```diff
@@ -25,85 +55,256 @@ import type {
+  Range,
+  Rangeable,
 } from "../aggregations";
+import { convertToAggregationResult } from "../aggregations/aggregationConverters";
 import type {
-  ActionResponse,
+  ArrayType,
   Attachment,
   AttachmentMetadata,
-  BaseObjectType,
+  AttachmentType,
+  BooleanType,
+  ByteType,
+  DateType,
+  DecimalType,
+  DoubleType,
   Edits,
+  FloatType,
+  GeoJson,
+  GeoJsonPoint,
+  GeoPointType,
+  GeoShapeType,
+  IntegerType,
+  LongType,
   ObjectSetDefinition,
+  ObjectSetType,
+  ObjectType,
   OntologyObject,
   OntologyType,
+  QueryBucketKey,
+  QueryBucketKeyType,
+  QueryBucketRangeableType,
+  QueryBucketValueType,
   QueryResponse,
+  QueryValueType,
+  RangeType,
+  SetType,
+  ShortType,
+  StringType,
+  StructType,
+  ThreeDimensionalAggregation,
+  ThreeDimensionalAggregationType,
   TimeSeriesPoint,
+  TimestampType,
+  TwoDimensionalAggregation,
+  TwoDimensionalAggregationType,
 } from "../baseTypes";
-import type { ParameterValue } from "../baseTypes/ParameterValue";
+import {
+  ActionResponse,
+  AttachmentProperty,
+  BaseObjectType,
+  GeoPoint,
+  GeoShape,
+  isOntologyObject,
+  LocalDate,
+  Timestamp,
+} from "../baseTypes";
+import type {
+  ParameterValue,
+  PrimitiveParameterValue,
+} from "../baseTypes/ParameterValue";
 import type { SearchClause } from "../filters";
 import type { Page } from "../paging";
+import {
+  AggregateObjectsErrorHandler,
+  AttachmentsErrorHandler,
+  ExecuteActionErrorHandler,
+  ExecuteQueryErrorHandler,
+  GetLinkedObjectErrorHandler,
+  GetObjectErrorHandler,
+  handleAggregateObjectsError,
+  handleAttachmentsError,
+  handleExecuteActionError,
+  handleExecuteQueryError,
+  handleGetLinkedObjectError,
+  handleGetObjectError,
+  handleListLinkedObjectsError,
+  handleListObjectsError,
+  handleLoadObjectSetError,
+  handleTimeSeriesError,
+  ListLinkedObjectsErrorHandler,
+  ListObjectsErrorHandler,
+  LoadObjectSetErrorHandler,
+  TimeSeriesErrorHandler,
+} from ".";
+import { mapAggregation, mapBucketing } from "./AggregationUtils";
 import type {
   ActionError,
   AggregateObjectsError,
   AttachmentsError,
+  FoundryApiError,
   GetLinkedObjectError,
   GetObjectError,
   ListLinkedObjectsError,
   ListObjectsError,
   LoadObjectSetError,
   QueryError,
   TimeSeriesError,
+  UnknownError,
 } from "./Errors";
 import type { OntologyMetadata } from "./OntologyMetadata";
 import type { OntologyObjectFactory } from "./OntologyObjectFactory";
-import type { Result } from "./Result";
+import {
+  iterateReadableStream,
+  parseStreamedResponse,
+} from "./parseStreamedResponse";
+import type { Err, Ok, Result } from "./Result";
 
 export class OntologyProvider {
-  private client: any;
-
   constructor(
     private authClient: Auth,
     private stack: string,
     private ontologyMetadata: OntologyMetadata,
     private objectFactory?: OntologyObjectFactory,
+    private fetchFn = globalThis.fetch,
+    // TODO contextPath for createOpenApiRequest?
```

Comment:
Do we need to support contextPath for this?

Reply:
I haven't seen people ask for it yet, but we probably should long term

Reply:
will rework in a flup pr